common/src/main/java/org/terracotta/angela/common/util/RetryUtils.java

```diff
@@ -48,50 +37,14 @@ public static boolean waitFor(Callable<Boolean> condition, long maxWaitTimeMilli
    */
   public static boolean waitFor(Callable<Boolean> condition, long maxWaitDuration, TimeUnit timeUnit) {
     TimeBudget timeBudget = new TimeBudget(maxWaitDuration, timeUnit);
-    long currRetryTime = min(maxWaitDuration, 100);
-    ExecutorService executorService = Executors.newSingleThreadExecutor(Thread::new);
     boolean success = false;
-
-    while (timeBudget.remaining() > 0 && !Thread.currentThread().isInterrupted()) {
```

Comment:
the implementation is flawed: in case of a short delay when spawning the executor is taking more time than the delay.